trio/_core/_tests/tutil.py

```diff
@@ -86,37 +86,26 @@ def _noop(*args, **kwargs):
     pass
 
 
-if sys.version_info >= (3, 8):
-
-    @contextmanager
-    def restore_unraisablehook():
-        sys.unraisablehook, prev = sys.__unraisablehook__, sys.unraisablehook
-        try:
-            yield
-        finally:
-            sys.unraisablehook = prev
-
-    @contextmanager
-    def disable_threading_excepthook():
-        if sys.version_info >= (3, 10):
-            threading.excepthook, prev = threading.__excepthook__, threading.excepthook
-        else:
-            threading.excepthook, prev = _noop, threading.excepthook
-
-        try:
-            yield
-        finally:
-            threading.excepthook = prev
+@contextmanager
+def restore_unraisablehook():
+    sys.unraisablehook, prev = sys.__unraisablehook__, sys.unraisablehook
+    try:
+        yield
+    finally:
+        sys.unraisablehook = prev
 
-else:
 
-    @contextmanager
-    def restore_unraisablehook():  # pragma: no cover
-        yield
+@contextmanager
+def disable_threading_excepthook():  # pragma: no cover # not used
+    if sys.version_info >= (3, 10):
+        threading.excepthook, prev = threading.__excepthook__, threading.excepthook
+    else:
+        threading.excepthook, prev = _noop, threading.excepthook
```

Comment:
This is imported in `trio/_core/_tests/test_thread_cache.py` but (surprise surprise!) not used at all. Should I just remove the function completely?

Reply:
I think so. While the `_tests` folder is technically still visible (cause of deprecation) and I got confused by that, this is actually the test folder under `_core` which is totally not visible!

tl;dr no stability promises for this file, feel free to remove. We can always revert if necessary.

Reply:
I was mostly thinking if it "should" be used in tests, or likely to be used in the future, such that we'd want it sticking around.

Reply:
Oh duh, sorry for missing that.

Looks like it was added for https://github.com/python-trio/trio/commit/fd45d4d8b7694715962b26db948417c02c6cab0b#diff-7ca9928b173566036b4074df65003dc54bd9cedc477b05cd8635ee703da19077R153 but that test was later changed in https://github.com/python-trio/trio/commit/14eb6fe8c2a30a9323ae4dc1901b8b0408c7c1e8 to have some form of it in the main library.

Looks like someone forgot to remove it, so this removal looks good.